dali/operators/reader/reader_op.h

```diff
@@ -179,9 +179,8 @@ class DataReader : public Operator<Backend> {
     // Consume batch
     Operator<Backend>::Run(ws);
     CUDA_CALL(cudaStreamSynchronize(ws.stream()));
-    for (int sample_idx = 0; sample_idx < max_batch_size_; sample_idx++) {
-      auto sample = MoveSample(sample_idx);
-    }
+    // clear samples from the current batch so they can be recycled
+    ClearCurentBatch();
```

Comment:
So, after `Run` this queue entry is empty, right? It's now the responsibility of `Prefetch` to resize it to the right size.
While you're at it, could you swap `reserve` and `clear` in `Prefetch`? Now `reserve` goes before `clear` which is rather counterproductive.

Reply:
> So, after Run this queue entry is empty, right?

Yes

> It's now the responsibility of Prefetch to resize it to the right size.

It always was but was used only during the first iterations. Now it happens always.

> While you're at it, could you swap reserve and clear in Prefetch? Now reserve goes before clear which is rather counterproductive.

Done.